fs/littlefs/lfs_vfs.c

```diff
@@ -183,6 +183,62 @@ static void littlefs_semgive(FAR struct littlefs_mountpt_s *fs)
   nxsem_post(&fs->sem);
 }
 
+/****************************************************************************
+ * Name: littlefs_convert_result
+ ****************************************************************************/
+
+static int littlefs_convert_result(int ret)
+{
+  switch (ret)
+    {
+      case LFS_ERR_IO:
+        return -EIO;
+
+      case LFS_ERR_CORRUPT:
+        return -EFAULT;
+
+      case LFS_ERR_NOENT:
+        return -ENOENT;
+
+      case LFS_ERR_EXIST:
+        return -EEXIST;
+
+      case LFS_ERR_NOTDIR:
+        return -ENOTDIR;
+
+      case LFS_ERR_ISDIR:
+        return -EISDIR;
+
+      case LFS_ERR_NOTEMPTY:
+        return -ENOTEMPTY;
+
+      case LFS_ERR_BADF:
+        return -EBADF;
+
+      case LFS_ERR_FBIG:
+        return -EFBIG;
+
+      case LFS_ERR_INVAL:
+        return -EINVAL;
+
+      case LFS_ERR_NOSPC:
+        return -ENOSPC;
+
+      case LFS_ERR_NOMEM:
+        return -ENOMEM;
+
+      case LFS_ERR_NOATTR:
+        return -ENODATA;
+
+      case LFS_ERR_NAMETOOLONG:
+        return -ENAMETOOLONG;
+
+      case LFS_ERR_OK:
+      default:
+        return ret;
```

Comment:
Should we
```
      case LFS_ERR_OK:
        return OK;

      default:
        return ret > 0 ? ret : ERROR;
```
Or we keep some of the error codes not mapped?

Reply:
This patch already covered all original error codes, and ERROR isn't every good error code, I think real return value is best value if out of range.